compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -233,6 +235,22 @@ public Symbol getBCompiledSymbol(BSymbol symbol, String name) {
             return createXMLNamespaceSymbol((BXMLNSSymbol) symbol);
         }
 
+        if (symbol.kind == SymbolKind.RESOURCE_PATH_IDENTIFIER_SEGMENT) {
+            return createResourceNamedPathSegment((BResourcePathSegmentSymbol) symbol);
+        }
+
+        if (symbol.kind == SymbolKind.RESOURCE_ROOT_PATH_SEGMENT) {
+            return createPathParamSymbol(symbol, PathSegment.Kind.ROOT_PATH);
```

Comment:
Dont we have tests to cover this line?

Reply:
Removed unnecessary dot-path symbol creation https://github.com/ballerina-platform/ballerina-lang/pull/39477/commits/d97348a8f990f74f0d34094d933a3d89a4542ab8

Reply:
Reason: we should not create a symbol for dot-path if we put the cursor position near the dot

```
funciton get <cursor>. ()
```